modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureClientProvider.java

```diff
@@ -33,7 +33,10 @@
 import com.azure.storage.blob.BlobServiceAsyncClient;
 import com.azure.storage.blob.BlobServiceClient;
 import com.azure.storage.blob.BlobServiceClientBuilder;
+import com.azure.storage.common.StorageSharedKeyCredential;
+import com.azure.storage.common.implementation.connectionstring.StorageAuthenticationSettings;
 import com.azure.storage.common.implementation.connectionstring.StorageConnectionString;
+import com.azure.storage.common.implementation.connectionstring.StorageEndpoint;
```

Comment:
FYI.
The fix for your problem will go out with next storage release.
See https://github.com/Azure/azure-sdk-for-java/issues/29998
https://github.com/Azure/azure-sdk-for-java/pull/30003

When the fix is out I suggest to not use types from `*.implementation.*` - we don't export them in our modules - which might be a problem on latest java runtime versions.

If there's anything else we can do to help here, please let us know.